### ISSUE TITLE
LFG: reward message is wrong when max level

### ISSUE THREAD

**Issue by qyh214** [TrinityCore/TrinityCore#25595]
**Description:**

LFG Frame show reward is wrong when player is max level.(xp should not show when player is max level)

**Expected behaviour:**

when player is max level,xp not show in lfg frame.

**Steps to reproduce the problem:**

1. in game and type I open lfg frame.

**Branch(es):**

both

**TC rev. hash/commit:** 

a183ca5bf5c424c4ac0940542a5b0942f242fb2e

**Operating system:** win 10

**Comment by Faq** [comment]
What lvl for you is max level? Could you add some picture?

**Comment by qyh214** [comment]
> 
> 
> What lvl for you is max level? Could you add some picture?

lvl 80, and I want to random lich king heric dungeon finder.
![20201021171006](https://user-images.githubusercontent.com/3896160/96698986-52478700-13c0-11eb-84ff-717ce656d938.jpg)

I remember not show xp at max level

**Comment by offl** [comment]
793 gold? Really?

**Comment by Javilico92** [comment]
![image](https://user-images.githubusercontent.com/60853949/96746599-a43fdb00-13c7-11eb-908a-0e1bdcc14adb.png)

Like this?

**Comment by illfated** [comment]
@Javilico92 : Yes, your window frame looks correct, also just as described by the ticket author as well.

**Comment by Javilico92** [comment]
Take a look at QuestDef.cpp in line 206 and add this
```c++
if (!player->IsMaxLevel())
            return 0;
```

**Comment by illfated** [comment]
(just to add a diff for the comment above)

```diff
diff --git a/src/server/game/Quests/QuestDef.cpp b/src/server/game/Quests/QuestDef.cpp
index ee767d034c..ccf0e69829 100644
--- a/src/server/game/Quests/QuestDef.cpp
+++ b/src/server/game/Quests/QuestDef.cpp
@@ -200,12 +200,15 @@ void Quest::LoadQuestMailSender(Field* fields)
 }
 
 uint32 Quest::GetXPReward(Player const* player) const
 {
     if (player)
     {
+        if (player->IsMaxLevel())
+            return 0;
+
         int32 quest_level = (_level == -1 ? player->GetLevel() : _level);
         QuestXPEntry const* xpentry = sQuestXPStore.LookupEntry(quest_level);
         if (!xpentry)
             return 0;
 
         int32 diffFactor = 2 * (quest_level - player->GetLevel()) + 20;
```

I could easily be mistaken about where the added lines need to be placed, so feel free to correct me on this.

**Comment by Javilico92** [comment]
> (just to add a diff for the comment above)
> 
> ```diff
> diff --git a/src/server/game/Quests/QuestDef.cpp b/src/server/game/Quests/QuestDef.cpp
> index ee767d034c..ccf0e69829 100644
> --- a/src/server/game/Quests/QuestDef.cpp
> +++ b/src/server/game/Quests/QuestDef.cpp
> @@ -200,12 +200,15 @@ void Quest::LoadQuestMailSender(Field* fields)
>  }
>  
>  uint32 Quest::GetXPReward(Player const* player) const
>  {
>      if (player)
>      {
> +        if (player->IsMaxLevel())
> +            return 0;
> +
>          int32 quest_level = (_level == -1 ? player->GetLevel() : _level);
>          QuestXPEntry const* xpentry = sQuestXPStore.LookupEntry(quest_level);
>          if (!xpentry)
>              return 0;
>  
>          int32 diffFactor = 2 * (quest_level - player->GetLevel()) + 20;
> ```
> 
> I could easily be mistaken about where the added lines need to be placed, so feel free to correct me on this.

Its nice. But how diff mark works?

**Comment by illfated** [comment]
> 
> Its nice. But how diff mark works?

Thank you for asking. This is about [writing on GitHub](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github), [MarkDown (MD)](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet), [GitHub Flavored MarkDown (GFM)](https://github.github.com/gfm/#introduction) and how MarkDown tags can be used to create code blocks, code block fencing and also be combined with [Code Syntax Highlighting](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/creating-and-highlighting-code-blocks#syntax-highlighting).

In this case, I used the 3 backticks (accent grave x3) combined with the word "diff" as code syntax, like this: **```diff**

**Comment by qyh214** [comment]
> 
> 
> 793 gold? Really?

@offl haha,I have set `Rate.Quest.Money.Reward`in conf.



> 
> 
> Take a look at QuestDef.cpp in line 206 and add this
> 
> ```c++
> if (player->IsMaxLevel())
>             return 0;
> ```

@Javilico92 thanks,I have test and seems well,maybe you can PR!
